guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/fork/ForkRepositoryPresenter.java

```diff
@@ -0,0 +1,203 @@
+import org.guvnor.structure.repositories.RepositoryService;
+import org.jboss.errai.bus.client.api.messaging.Message;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.ErrorCallback;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.mvp.impl.DefaultPlaceRequest;
+import org.uberfire.rpc.SessionInfo;
+import org.uberfire.security.authz.AuthorizationManager;
+
+import static org.guvnor.structure.security.RepositoryFeatures.*;
+
+@Dependent
+public class ForkRepositoryPresenter implements ForkRepositoryView.Presenter {
+
+    private RepositoryPreferences repositoryPreferences;
+
+    private ForkRepositoryView view;
+
+    private Caller<RepositoryService> repositoryService;
+
+    private Caller<OrganizationalUnitService> organizationalUnitService;
+
+    private PlaceManager placeManager;
+
+    private AuthorizationManager authorizationManager;
+
+    private Map<String, OrganizationalUnit> availableOrganizationalUnits = new HashMap<String, OrganizationalUnit>();
+
+    private SessionInfo sessionInfo;
+    private boolean assetsManagementIsGranted = false;
+
+    @Inject
+    public ForkRepositoryPresenter( final RepositoryPreferences repositoryPreferences,
+                                    final ForkRepositoryView view,
+                                    final Caller<RepositoryService> repositoryService,
+                                    final Caller<OrganizationalUnitService> organizationalUnitService,
+                                    final PlaceManager placeManager,
+                                    final AuthorizationManager authorizationManager,
+                                    final SessionInfo sessionInfo ) {
+        this.repositoryPreferences = repositoryPreferences;
+        this.view = view;
+        this.repositoryService = repositoryService;
+        this.organizationalUnitService = organizationalUnitService;
+        this.placeManager = placeManager;
+        this.authorizationManager = authorizationManager;
+        this.sessionInfo = sessionInfo;
+    }
+
+    @PostConstruct
+    public void init() {
+        view.init( this,
+                   isOuMandatory() );
+        setAssetsManagementGrant();
+    }
+
+    @Override
+    public void handleCancelClick() {
+        view.hide();
+    }
+
+    @Override
+    public void handleForkClick() {
+        repositoryService.call( getNormalizeRepositoryNameCallback() )
+                .normalizeRepositoryName( view.getTargetOrganizationalUnitSelected() );
+    }
+
+    private RemoteCallback<String> getNormalizeRepositoryNameCallback() {
+        return new RemoteCallback<String>() {
+            @Override
+            public void callback( final String normalizedName ) {
+                lockScreen();
+
+                final String scheme = "git";
+                final String[] source = view.getSourceSelectedOrganizationalUnit().split( "/" );
```

Comment:
I did not see any validation of either "source Organisational Unit" nor" target Organisational Unit". Since both are ```TextBox``` in the UI; is it not possible that either could *not* contain a "/" and hence the split result in arrays of size one leading to the below use of index 1 to fail?